lib/pbench/agent/tool_meister.py

```diff
@@ -248,162 +348,133 @@ def wait(self):
-                "--socket=./pmcd.socket",
-                "--port=55677",
-                f"--config={self.pbench_install_dir}/templates/pmcd.conf",
-            ]
-        else:
-            self.pmcd_args = None
-        pmlogger_path = find_executable("pmlogger")
-        if pmlogger_path:
-            self.pmlogger_args = [
-                pmlogger_path,
-                "--log=-",
-                "--report",
-                "-t",
-                "3s",
-                "-c",
-                f"{self.pbench_install_dir}/templates/pmlogger.conf",
-                "--host=localhost:55677",
-                f"{self.tool_dir}/%Y%m%d.%H.%M",
-            ]
-        else:
-            self.pmlogger_args = None
+            # FIXME - Shouldn't this be provided by the environment?
+            os.environ["PATH"] += ":/usr/libexec/pcp/bin"
+        self.pmcd_args = None
         self.pmcd_process = None
+        self.pmlogger_args = None
         self.pmlogger_process = None
-        if self.tool_dir:
-            try:
-                self.tool_dir.mkdir()
-            except Exception as exc:
-                self.logger.error(
-                    "Failed to create tool directory '%s': '%s'", self.tool_dir, exc
-                )
 
     def install(self):
-        if self.pmcd_args is None:
+        pmcd_path = find_executable("pmcd")
+        if not pmcd_path:
             return (1, "pcp tool (pmcd) not found")
-        elif self.pmlogger_args is None:
+        self.pmcd_args = [
+            pmcd_path,
+            "--foreground",
+            "--socket=./pmcd.socket",
+            "--port=55677",
+            f"--config={self.pbench_install_dir}/templates/pmcd.conf",
+        ]
+
+        pmlogger_path = find_executable("pmlogger")
+        if not pmlogger_path:
             return (1, "pcp tool (pmlogger) not found")
+        self.pmlogger_args = [
+            pmlogger_path,
+            "--log=-",
+            "--report",
+            "-t",
+            "3s",
+            "-c",
+            f"{self.pbench_install_dir}/templates/pmlogger.conf",
+            "--host=localhost:55677",
+            f"{self.tool_inst_dir}/%Y%m%d.%H.%M",
+        ]
+
         return (0, "pcp tool (pmcd and pmlogger properly installed")
 
     def start(self):
-        assert self.tool_dir is not None, "Logic bomb!  no tool directory provided!"
-        if self.pmcd_process or self.pmlogger_process:
+        if self.pmcd_process is not None:
+            raise ToolException(
+                f"Tool({self.name}) has an unexpected pmcd process running"
+            )
+        if self.pmlogger_process is not None:
             raise ToolException(
-                f"Tool({self.name}) has an unexpected process still running"
+                f"Tool({self.name}) has an unexpected pmlogger process running"
             )
 
         self.logger.info(
-            "%s: start_tool -- %s -- %s",
+            "%s: start_tool -- '%s' && '%s'",
```

Comment:
As long as you're getting rid of `--`'s, I would get rid of them both.  E.g.,
```
        self.logger.info(
            "%s: [start_tool] '%s' && '%s'",
```

Reply:
That would make them really different from all the others.